examples/firewall/firewall.c

```diff
@@ -0,0 +1,332 @@
+        printf("Usage: %s [EAL args] -- [NF_LIB args] -- -p <print_delay>\n\n", progname);
+}
+
+/*
+ * Parse the application arguments.
+ */
+static int
+parse_app_args(int argc, char *argv[], const char *progname) {
+        int c, dst_flag = 0, rules_init = 0;
+
+        while ((c = getopt (argc, argv, "d:f:p:b")) != -1) {
+                switch (c) {
+                case 'd':
+                        destination = strtoul(optarg, NULL, 10);
+                        dst_flag = 1;
+                        break;
+                case 'p':
+                        RTE_LOG(INFO, APP, "print_delay = %d\n", 0);
+                        break;
+
+                case 'f':
+                        rule_file = malloc(sizeof(char) * (strlen(optarg)));
+                        strcpy(rule_file, optarg);
+                        rules_init = 1;
+                        break;
+                case 'b':
+                        RTE_LOG(INFO, APP, "Debug mode enabled; printing the source IP addresses of each incoming packet as well as drop/forward status\n");
+                        debug = 1;
+                        break;
+                case '?':
+                        usage(progname);
+                        if (optopt == 'p')
+                                RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                        if (optopt == 'd')
+                                RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                        if (optopt == 'f')
+                                RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                        else if (isprint(optopt))
+                                RTE_LOG(INFO, APP, "Unknown option `-%c'.\n", optopt);
+                        else
+                                RTE_LOG(INFO, APP, "Unknown option character `\\x%x'.\n", optopt);
+                        return -1;
+                default:
+                        usage(progname);
+                        return -1;
+                }
+        }
+
+        if (!dst_flag) {
+            RTE_LOG(INFO, APP, "Firewall NF requires a destination NF with the -d flag.\n");
+            return -1;
+        }
+        if (!debug) {
+            RTE_LOG(INFO, APP, "Running normal mode, use -b flag to enable debug mode\n");
+        }
+        if (!rules_init) {
+            RTE_LOG(INFO, APP, "Please specify a rules JSON file with -f FILE_NAME\n");
+            return -1;
+        }
+        return optind;
+}
+
+
+static int
+packet_handler(struct rte_mbuf* pkt, struct onvm_pkt_meta* meta, __attribute__((unused)) struct onvm_nf_info* nf_info) {
+        struct ipv4_hdr* ipv4_hdr;
+        uint32_t rule = 0;
+        uint32_t track_ip = 0;
+
+        if(onvm_pkt_is_ipv4(pkt)){
+                ipv4_hdr = onvm_pkt_ipv4_hdr(pkt);
+
+                int no_result = rte_lpm_lookup(lpm_tbl, ipv4_hdr->src_addr, &rule);
+                if(!no_result){
```

Comment:
Same as above, instead of
```
if (something_good)
    handle_success
else
    handle_failure
```
preffer
```
if (something_not_good)
    handle_failure

handle_success
```